UPGRADE.md

```diff
@@ -34,6 +39,126 @@ As part of the update of flysystem the support for the guzzle client package `gu
 
 The `GoogleGeolocator` and the `NominatimGeolocator` no longer support the Guzzle client and require a `Symfony\HttpClient` client instead.
 
+### Updating to flysystem 3
+
+The Sulu media storage uses now the Flysystem Bundle which need to be configured:
+
+```yaml
+# config/packages/flysystem.yaml
+flysystem:
+    storages:
+        default.storage:
+            adapter: 'local'
+            options:
+                directory: '%kernel.project_dir%/var/uploads'
+```
+
+Here are some examples on how to migrate individual providers:
+
+<details>
+  <summary>Sulu Media S3 to Flysystem S3</summary>
+
+Before:
+
+```yaml
+sulu_media:
+    storage: s3
+    storages:
+        s3:
+            key: 'your aws s3 key'
+            secret: 'your aws s3 secret'
+            bucket_name: 'your aws s3 bucket name'
+            path_prefix: 'optional/path/prefix'
+            region: 'eu-west-1'
+```
+
+New:
+
+```yaml
+flysystem:
+    storages:
+        default.storage:
+            adapter: 'aws'
+            options:
+                client: 'aws_client_service' # The service ID of the Aws\S3\S3Client instance
```

Comment:
Do they need to configure that in some kind of a way / create a service definition there self or does the `aws_client_service` already exist out of the box? 

If they need to create it there should be also documented that step for the uprade:

```yaml
services:
    aws_client_service
        class: Aws\S3\S3Client
```

Reply:
Well that's a good question. 
Well first of all, if you don't install it explicitly it won't be there. (So by default the s3 bundle will no longer be installed.)

But it looks like the default service is not configured. However, I don't know how it was configured before so it might be that it is set up if you used to use the S3 integration. Same goes for the other integrations.

Reply:
Okay I've dug a little S3 is the only storage which Sulu used to set up with a compiler pass (now deleted). I've added the new service definition to the upgrade guide. As for the other storage solutions, the leauge bundles do not come with automatic configs so you have to set those services up yourself.